modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryTableStorage.java

```diff
@@ -0,0 +1,261 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.pagememory;
+
+import static org.apache.ignite.internal.pagememory.PageIdAllocator.FLAG_AUX;
+import static org.apache.ignite.internal.storage.StorageUtils.groupId;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.ignite.configuration.schemas.table.TableConfiguration;
+import org.apache.ignite.configuration.schemas.table.TableView;
+import org.apache.ignite.internal.pagememory.FullPageId;
+import org.apache.ignite.internal.pagememory.evict.PageEvictionTrackerNoOp;
+import org.apache.ignite.internal.pagememory.metric.IoStatisticsHolderNoOp;
+import org.apache.ignite.internal.pagememory.persistence.PageMemoryImpl;
+import org.apache.ignite.internal.pagememory.persistence.checkpoint.CheckpointTimeoutLock;
+import org.apache.ignite.internal.pagememory.persistence.store.FilePageStore;
+import org.apache.ignite.internal.pagememory.util.PageLockListenerNoOp;
+import org.apache.ignite.internal.storage.StorageException;
+import org.apache.ignite.internal.storage.pagememory.io.PartitionMetaIo;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * Implementation of {@link AbstractPageMemoryTableStorage} for persistent case.
+ */
+class PersistentPageMemoryTableStorage extends AbstractPageMemoryTableStorage {
+    /**
+     * Constructor.
+     *
+     * @param tableCfg Table configuration.
+     * @param dataRegion Data region for the table.
+     */
+    public PersistentPageMemoryTableStorage(
+            TableConfiguration tableCfg,
+            PersistentPageMemoryDataRegion dataRegion
+    ) {
+        super(tableCfg, dataRegion);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void start() throws StorageException {
+        super.start();
+
+        TableView tableView = tableCfg.value();
+
+        try {
+            // TODO: IGNITE-16665 Directory name needs to be corrected to support table renaming
+            ((PersistentPageMemoryDataRegion) dataRegion)
+                    .filePageStoreManager()
+                    .initialize(tableView.name(), groupId(tableView), tableView.partitions());
+        } catch (IgniteInternalCheckedException e) {
+            throw new StorageException("Error initializing file page stores for table: " + tableView.name(), e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected VolatilePageMemoryPartitionStorage createPartitionStorage(int partId) throws StorageException {
```

Comment:
Shouldn't it be PersistentPageMemoryPartitionStorage?

Reply:
Fix it.